src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -219,117 +207,103 @@ def _entity_ids_by_type(self,
+                     area
+        :param tmpdir: The path to a local directory where the repository will
+                       be cloned
+
         """
-        token = os.environ['GITHUB_TOKEN']  # A GitHub personal access token
-        log.debug('Requesting GitHub repo %s', (owner, name, ref, path))
-        github_api = Github(token)
-        rate_limit = github_api.get_rate_limit()
-        log.debug('GitHub API rate limit limit=%i, remaining=%i, reset=%s',
-                  rate_limit.core.limit, rate_limit.core.remaining, rate_limit.core.reset)
-        repo = github_api.get_repo(f'{owner}/{name}')
-        path = () if path == '' else tuple(path.split('/'))
-        return cls(repo=repo, ref=ref, path=path)
+        log.debug("Cloning '%s' into '%s'", url, tmpdir)
+        repo = git.Repo.clone_from(url, tmpdir)
+        commit = repo.commit(ref)
+        tree = repo.tree(commit)
+        tree = cls._get_subtree(tree, path)
+        return cls(url=url, repo=repo, commit=commit, path=path, tree=tree)
 
     @classmethod
-    def from_url(cls, url: str) -> 'GitHubStagingAreaFactory':
-        """
-        :param url: The URL of a staging area in a GitHub repository with syntax
-                    `https://github.com/<OWNER>/<NAME>/tree/<REF>[/<PATH>]`.
-                    Note that REF can be a branch, tag, or commit SHA. If REF
-                    contains special characters like `/`, '?` or `#` they must
-                    be URL-encoded. This is especially noteworthy for `/` since
-                    it's the only way to distinguish slashes in REF from those
-                    in PATH. The slashes in PATH must not be URL-encoded, while
-                    occurrences of `#` and `?` must.
-        """
-        parsed_url = furl(url)
-        require(parsed_url.scheme == 'https', url)
-        require(parsed_url.host == 'github.com', url)
-        require(len(parsed_url.path.segments) > 3, url)
-        require(parsed_url.path.segments[2] == 'tree', url)
-        owner, name = parsed_url.path.segments[0:2]
-        ref = parsed_url.path.segments[3]
-        path = '/'.join(parsed_url.path.segments[4:])
-        return cls.create(owner=owner, name=name, ref=ref, path=path)
+    def _get_subtree(cls, root: git.Tree, path: tuple[str, ...]) -> git.Tree:
+        while len(path) > 0:
+            p, *path = path
+            for tree in root.trees:
+                if tree.name == p:
+                    root = tree
+                    break
+            else:
+                raise ValueError(f'Could not find path {p!r} in {root.path!r}')
+        return root
 
     def load_staging_area(self) -> StagingArea:
-        staging_area_folders = self._get_folders(path=self.path)
+        staging_area_folders = {t.name for t in self.tree.trees}
         expected_folders = {'data', 'descriptors', 'links', 'metadata'}
         require(set(staging_area_folders) == expected_folders,
```

Comment:
Looks like it's already a `set`
```suggestion
        require(staging_area_folders == expected_folders,
```

Also, doesn't the expected value usually go first?